### PR TITLE
Add `edit` context to patterns REST controllers

### PR DESCRIPTION
**Pull request by ntsekouras**
We need to add `edit` context to the patterns REST controllers as it has changed in the backport to core: https://github.com/WordPress/wordpress-develop/pull/2567.

This also adds a check about the theme's `patterns` directory being readable and changes `errors` to `_doing_it_wrong` for pattern registration from this theme's directory(patterns).

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
Awesome, thank you for doing it.

In WordPress core I fight with some failing unit tests after similar changes. We should learn if that's the case here as well once CI jobs finish.

**Comment by gziolo** [comment]
I don't think we need to backport anything back to WP core from this PR - removed the label.

File: lib/compat/wordpress-6.0/block-patterns.php [hunk 1]
```diff
@@ -98,31 +98,38 @@ function gutenberg_register_theme_block_patterns() {
 
 	foreach ( $themes as $theme ) {
 		$dirpath = $theme->get_stylesheet_directory() . '/patterns/';
+		if ( ! is_dir( $dirpath ) || ! is_readable( $dirpath ) ) {
+			continue;
+		}
 		if ( file_exists( $dirpath ) ) {
```
Comment (jsnajdr) [inline lib/compat/wordpress-6.0/block-patterns.php]: nit: is this `file_exists` check still necessary after the "`$dirpath` is a readable directory" check has passed?

**Review by jsnajdr — APPROVED** [review]
:+1:

**Pull request merged**